File: output/qwen_2.5_72b_instruct/avatar/direct_translation/java/python/successful_translation/codeforces_637_A.py

import sys
from collections import Counter

n = int(input())
ara = list(map(int, input().split()))
map = Counter(ara)
max_val = max(map.values())
mm = {}

for i in range(n):
    if ara[i] not in mm:
        mm[ara[i]] = 1
        if mm[ara[i]] == max_val:
            print(ara[i])
            break
    else:
        mm[ara[i]] += 1
        if mm[ara[i]] == max_val:
            print(ara[i])
            break

File: output/qwen_2.5_72b_instruct/avatar/direct_translation/python/java/successful_translation/codeforces_637_A.java
import java.util.HashMap;
import java.util.Scanner;

public class codeforces_637_A {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int n = input.nextInt();
        input.nextLine();
        String[] s = input.nextLine().trim().split(" ");
        HashMap<String, Integer> w = new HashMap<>();
        for (String i : s) {
            w.put(i, w.getOrDefault(i, 0) + 1);
        }
        String[] reversed = new String[s.length];
        for (int i = 0; i < s.length; i++) {
            reversed[i] = s[s.length - 1 - i];
        }
        int c = -1;
        String a = "";
        for (String i : w.keySet()) {
            if (w.get(i) == w.values().stream().max(Integer::compare).get()) {
                int index = java.util.Arrays.asList(reversed).indexOf(i);
                if (index > c) {
                    a = i;
                    c = index;
                }
            }
        }
        System.out.println(a);
    }
}
